### PR TITLE
feat: environment variables token name inconsistency

### LINKED ISSUE #2652: discuss: Inconsistency in GitHub/GitLab environment variable names
In our `app-config.yml`, we seem to have variants of GitLab/GitHub environment variables  -

1. https://github.com/spotify/backstage/blob/f9fa53153e822042b76f34fb894541cf713e9cc9/app-config.yaml#L79
2. https://github.com/spotify/backstage/blob/f9fa53153e822042b76f34fb894541cf713e9cc9/app-config.yaml#L133
3. https://github.com/spotify/backstage/blob/f9fa53153e822042b76f34fb894541cf713e9cc9/app-config.yaml#L140
4. https://github.com/spotify/backstage/blob/f9fa53153e822042b76f34fb894541cf713e9cc9/app-config.yaml#L102

* Can we pick one and make these consistent?
* How should we avoid such inconsistencies? Should there be one source of truth for all the environment variables instead of `app-config.yml`?

### PR DESCRIPTION
**Pull request by Marvin9**
## Hey, I just made a Pull Request!

Closes #2652 

```
GITHUB_PRIVATE_TOKEN & GITHUB_ACCESS_TOKEN => GITHUB_TOKEN
GITLAB_PRIVATE_TOKEN, GITLAB_ACCESS_TOKEN & SCAFFOLDER_GITLAB_PRIVATE_TOKEN => GITLAB_TOKEN
AZURE_PRIVATE_TOKEN => AZURE_TOKEN
GHE_PRIVATE_TOKEN => GHE_TOKEN
```

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by OrkoHunter** [comment]
Thanks for doing this! This is a breaking change. Let's use [changesets](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets) to document this change.

**Comment by Marvin9** [comment]
@OrkoHunter Updated. Let me know if this is sufficient or not.

**Review by OrkoHunter — COMMENTED** [review]
I am +1 for this. But expert supervision from @spotify/backstage-core is required here. 🙇 😅

File: .changeset/plenty-dolls-add.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+'@backstage/cli': major
+'@backstage/create-app': major
```
Comment (Rugvip) [inline .changeset/plenty-dolls-add.md:3]: This should be a minor change to any of the packages that read the env var directly, and a patch change for create-app. I doesn't really touch the cli or e2e-test either, since it's just dependency updates that will be reflected anyway.

File: packages/core/src/layout/HeaderTabs/index.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const HeaderTabs: React.FC<{
         {tabs.map((tab, index) => (
           <Tab
             label={tab.label}
-            key={tab.id}
+            key={`${tab.id}_${index}`}
```
Comment (Rugvip) [inline packages/core/src/layout/HeaderTabs/index.tsx:81]: This doesn't seem correct, would expect the user of the component needs to be changed instead? If it's a fix for a warning it'd be good to submit this as a separate PR!

File: .changeset/plenty-dolls-add.md (hunk 1 above)
Comment (Marvin9) [reply .changeset/plenty-dolls-add.md]: Thank you so much. It directly affects only ```techdocs-backend```.

**Review by OrkoHunter — APPROVED** [review]

**Review by emmaindal — APPROVED** [review]

**Review by marcuseide — APPROVED** [review]
